### PR TITLE
New package: nomic.gpt4all version 4.5.2.0

### PR DESCRIPTION
**Pull request by fasterinnerlooper**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/122133)

### CODE REVIEW

**Comment by sjetha** [comment]
@microsoft-github-policy-service agree

**Comment by fasterinnerlooper** [comment]
@microsoft-github-policy-service agree

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 2:56:15 PM:  > 2023-10-23 14:56:15.567 [CLI ] Package hash verification failed. SHA256 in manifest [78d0295e03004f69c2adab9e70dc091dd763d9c48627553f06d1aed002605c3a] does not match download [ab5bde3738604d2fbd4953d6f9218c8776c41c96b8ab7a212ab71c3ae5c3052f]

File: manifests/n/nomic/gpt4all/4.5.2.0/nomic.gpt4all.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.5.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: nomic.gpt4all
+PackageVersion: 4.5.2.0
+InstallerType: exe
+Installers:
+- InstallerUrl: https://gpt4all.io/installers/gpt4all-installer-win64.exe
+  Architecture: x64
+  InstallerSha256: 78D0295E03004F69C2ADAB9E70DC091DD763D9C48627553F06D1AED002605C3A
```
Comment (stephengillie) [inline manifests/n/nomic/gpt4all/4.5.2.0/nomic.gpt4all.installer.yaml:10]:
```suggestion
  InstallerSha256: AB5BDE3738604D2FBD4953D6F9218C8776C41C96B8AB7A212AB71C3AE5C3052F
```

**Comment by stephengillie** [comment]
Automatic Validation ended with:
image

The installer window appears to be behind the automation management window.

**Comment by stephengillie** [comment]
Found querying with `--?` gave a listing of switches:

```
Usage: C:\Users\User\AppData\Local\Temp\WinGet\nomic.gpt4all.4.5.2.0\gpt4all-installer-win64.exe [options] command <args> <key=value>

Qt Installer Framework supports both GUI and headless mode. The installation operations can be invoked with the following commands and options. Note that the options marked with "CLI" are available in the headless mode only.

Commands:
  in, install - install default or selected packages - <pkg ...>
  ch, check-updates - show available updates information on maintenance tool
  up, update - update all or selected packages - <pkg ...>
  rm, remove - uninstall packages and their child components - <pkg ...>
  li, list - list currently installed packages - <regexp>
  se, search - search available packages - <regexp>
    Note: The --filter-packages option can be used to specify
    additional filters for the search operation
  co, create-offline - create offline installer from selected packages - <pkg ...>
  cc, clear-cache - clear contents of the local metadata cache
  pr, purge - uninstall all packages and remove entire program directory

Options:
  -?, -h, --help                                 Displays help on commandline
                                                 options.
  --help-all                                     Displays help including Qt
                                                 specific options.
  -v, --version                                  Displays version information.
  -d, --verbose                                  Verbose mode. Prints out more
                                                 information. Adding -d or
                                                 --verbose more than once
                                                 increases verbosity.
  -g, --logging-rules <rules>                    Enables logging according to
                                                 passed rules. Comma separated
                                                 logging rules have the
                                                 following syntax:
                                                 loggingCategory=true/false.
                                                 Passing empty logging rules
                                                 enables all logging categories.
                                                 The following rules enable a
                                                 single category: ifw.*=false,
                                                 ifw.category=true. The
                                                 following logging categories
                                                 are available:
                                                 ifw.installer.installlog
                                                 ifw.server
  --ar, --add-repository <URI,...>               Add a local or remote
                                                 repository to the list of user
                                                 defined repositories.
  --at, --add-temp-repository <URI,...>          Add a local or remote
                                                 repository to the list of
                                                 temporary available
                                                 repositories.
  --st, --set-temp-repository <URI,...>          Set a local or remote
                                                 repository as temporary
                                                 repository, it is the only one
                                                 used during fetch.
                                                 Note: URI must be prefixed with
                                                 the protocol, i.e. file:///,
                                                 https://, http:// or ftp://.
  --sp, --system-proxy                           Use system proxy on Windows
                                                 and Linux. This option has no
                                                 effect on macOS. (Default)
  --np, --no-proxy                               Do not use system proxy.
  --su, --start-updater                          Start application in updater
                                                 mode. This will override the
                                                 internal marker that is used to
                                                 distinguish which kind of
                                                 binary is currently running.
  --sm, --start-package-manager                  Start application in package
                                                 manager mode. This will
                                                 override the internal marker
                                                 that is used to distinguish
                                                 which kind of binary is
                                                 currently running.
  --sr, --start-uninstaller                      Start application in
                                                 uninstaller mode. This will
                                                 override the internal marker
                                                 that is used to distinguish
                                                 which kind of binary is
                                                 currently running.
  -t, --root <directory>                         [CLI] Set installation root
                                                 directory.
  --oi, --offline-installer-name <filename>      [CLI] Set custom filename for
                                                 the generated offline
                                                 installer. Without this the
                                                 original filename is used with
                                                 an added "_offline-yyyy-MM-dd"
                                                 suffix.
  -p, --platform <plugin>                        Use the specified platform
                                                 plugin.
  --nf, --no-force-installations                 Allow deselecting components
                                                 that are marked as forced.
  --nd, --no-default-installations               Deselects components that are
                                                 marked as default.
  --ns, --no-size-checking                       Disable checking of free space
                                                 for installation target.
  --sv, --show-virtual-components                Show virtual components in
                                                 installer and package manager.
  -i, --install-compressed-repository <URI,...>  Installs QBSP or 7z file. The
                                                 QBSP (Board Support Package)
                                                 file must be a .7z file which
                                                 contains a valid repository.
  --cl, --create-local-repository                Create a local repository
                                                 inside the installation
                                                 directory. This option has no
                                                 effect on online installers.
  --fp, --filter-packages <element=regex,...>    [CLI] Comma separated list of
                                                 additional key-value pair
                                                 filters used to query packages
                                                 with the search command. The
                                                 keys can be any of the possible
                                                 package information elements,
                                                 like "DisplayName" and
                                                 "Description".
  --cp, --cache-path <path>                      Sets the path used for local
                                                 metadata cache. The path must
                                                 be writable by the current
                                                 user.
  --am, --accept-messages                        [CLI] Accepts all message
                                                 queries without user input.
  --rm, --reject-messages                        [CLI] Rejects all message
                                                 queries without user input.
  --aa, --auto-answer <identifier=value>         [CLI] Automatically answers
                                                 the message queries with the
                                                 message identifier and button
                                                 value. Several identifier=value
                                                 pairs can be given separated
                                                 with comma, for example
                                                 --auto-answer
                                                 message.id=Ok,message.id2=Cance
                                                 l.
  --da, --default-answer                         [CLI] Automatically answers to
                                                 message queries with their
                                                 default values.
  --al, --accept-licenses                        [CLI] Accepts all licenses
                                                 without user input.
  --file-query <identifier=value>                [CLI] Automatically sets the
                                                 QFileDialog values
                                                 getExistingDirectory() or
                                                 getOpenFileName() requested by
                                                 install script. Several
                                                 identifier=value pairs can be
                                                 given separated with comma, for
                                                 example --file-query
                                                 filedialog.id=C:Temp,filedialog
                                                 .id2=C:Temp2
  -c, --confirm-command                          [CLI] Confirms starting of
                                                 installation, update or removal
                                                 of components without user
                                                 input.
  -s, --script <file>                            Execute the script given as
                                                 argument.
  --ss, --start-server <mode, socketname, key>   Starts the application as
                                                 headless process waiting for
                                                 commands to execute. Mode can
                                                 be DEBUG or PRODUCTION. In
                                                 DEBUG mode, the option values
                                                 can be omitted. Note: The
                                                 server will not shutdown on his
                                                 own, you need to quit the
                                                 process by hand.
  --sc, --start-client <socketname, key>         Starts the application to
                                                 debug the client-server
                                                 communication. If a value is
                                                 omitted, the client will use a
                                                 default instead. Note: The
                                                 server process is not started
                                                 by the client application in
                                                 that case, you need to start it
                                                 on your own.
  -q, --squish-port <port number>                Give a port where Squish can
                                                 connect to. If no port is
                                                 given, attach to squish not
                                                 done. Note: To enable Squish
                                                 support you first need to build
                                                 IFW with SQUISH_PATH parameter
                                                 where SQUISH_PATH is pointing
                                                 to your Squish installation
                                                 folder: <path_to_qt>/bin/qmake
                                                 -r SQUISH_PATH=<pat_to_squish>
  --mco, --max-concurrent-operations <threads>   Specifies the maximum number
                                                 of threads used to perform
                                                 concurrent operations in the
                                                 unpacking phase of components.
                                                 Set to a positive number, or 0
                                                 (default) to let the
                                                 application determine the ideal
                                                 thread count from the amount of
                                                 logical processor cores in the
                                                 system.

Arguments:
  Command                                        Command to be run by
                                                 installer. Note that this
                                                 feature may be disabled by
                                                 vendor.
  Args                                           Extra arguments for command,
                                                 each separated by space.
  InstallerValue                                 Key-value pair to be set
                                                 internally by the framework.


```

**Comment by stephengillie** [comment]
Hi @fasterinnerlooper,

Does this package have silent installer switches? The switches above that I got from the client don't result in an unattended install.

**Comment by fasterinnerlooper** [comment]
Hi @stephengillie,

I've actually been looking into this as and when I get time, and I think I've got a silent installation process working. I'll update this PR with the changes.
I also didn't want to waste too much of your team's time on this if I wasn't pretty sure that I had it working properly, so apologies for not keeping you up to date on this.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 10:38:33 AM:  > 2023-10-24 10:38:32.918 [CLI ] Package hash verification failed. SHA256 in manifest 
[ab5bde3738604d2fbd4953d6f9218c8776c41c96b8ab7a212ab71c3ae5c3052f] does not match download 
[afcfd38dc06e2c368ff8c177c75b660402b42225f4d70c8825b45f954b7978a2]

File: manifests/n/nomic/gpt4all/4.5.2.0/nomic.gpt4all.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+PackageIdentifier: nomic.gpt4all
+PackageVersion: 4.5.2.0
+InstallerType: exe
+InstallerSwitches:
+  Silent: install --accept-licenses --confirm-command --am
+Installers:
+- InstallerUrl: https://gpt4all.io/installers/gpt4all-installer-win64.exe
+  Architecture: x64
+  InstallerSha256: AB5BDE3738604D2FBD4953D6F9218C8776C41C96B8AB7A212AB71C3AE5C3052F
```
Comment (stephengillie) [inline manifests/n/nomic/gpt4all/4.5.2.0/nomic.gpt4all.installer.yaml:12]:
```suggestion
  InstallerSha256: AFCFD38DC06E2C368FF8C177C75B660402B42225F4D70C8825B45F954B7978A2
```

**Comment by fasterinnerlooper** [comment]
Hi @stephengillie,

I double-checked the hash and it appears to be correct. Are you able to verify this?

**Review by stephengillie — APPROVED** [review]

**Comment by stephengillie** [comment]
> Hi @stephengillie,
> 
> I double-checked the hash and it appears to be correct. Are you able to verify this?

Yep, looks good to me.

**Pull request merged**